internal/templates/hosted-cluster/template.go

```diff
@@ -0,0 +1,337 @@
+  certPolicyController:
+    enabled: false
+  policyController:
+    enabled: true
+  searchCollector:
+    enabled: false`
+
+const ManagedCluster = `apiVersion: cluster.open-cluster-management.io/v1
+kind: ManagedCluster
+metadata:
+  name: "{{ .Spec.ClusterName }}"
+  annotations:
+    siteconfig.open-cluster-management.io/sync-wave: "2"
+    import.open-cluster-management.io/hosting-cluster-name: local-cluster
+    import.open-cluster-management.io/klusterlet-deploy-mode: Hosted
+    open-cluster-management/created-via: hypershift
+  labels:
+    cloud: BareMetal
+    vendor: OpenShift
+    ishosted: "true"
+spec:
+  hubAcceptsClient: true`
+
+// nolint:gosec
+const SshPubKeySecret = `apiVersion: v1
+kind: Secret
+metadata:
+  name: sshkey-{{ .Spec.ClusterName }}
+  namespace: "{{ .Spec.ClusterName }}"
+  labels:
+    hypershift.openshift.io/safe-to-delete-with-cluster: "true"
+  annotations:
+    siteconfig.open-cluster-management.io/sync-wave: "1"
+stringData:
+  id_rsa.pub: "{{ .Spec.SSHPublicKey }}"
+`
+
+/*
+  Node-level installation templates
+*/
+
+const InfraEnv = `apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  annotations:
+    siteconfig.open-cluster-management.io/sync-wave: "1"
+{{ if .SpecialVars.CurrentNode.HostRef }}
+  name: "{{ .SpecialVars.CurrentNode.HostRef.Name }}"
+  namespace: "{{ .SpecialVars.CurrentNode.HostRef.Namespace }}"
+{{ else }}
+  name: "{{ .SpecialVars.CurrentNode.HostName }}"
+  namespace: "{{ .Spec.ClusterName }}"
+{{ end }}
+spec:
+  sshAuthorizedKey: "{{ .Spec.SSHPublicKey }}"
+{{ if .Spec.Proxy }}
+  proxy:
+{{ .Spec.Proxy | toYaml | indent 4 }}
+{{ end }}
+{{ if .SpecialVars.CurrentNode.CPUArchitecture }}
+  cpuArchitecture: "{{ .SpecialVars.CurrentNode.CPUArchitecture }}"
+{{ else if and
+    (.Spec.CPUArchitecture)
+    (ne .Spec.CPUArchitecture "multi")
+}}
+  cpuArchitecture: "{{ .Spec.CPUArchitecture }}"
+{{ end }}
+  pullSecretRef:
+    name: "{{ .Spec.PullSecretRef.Name }}"
+  ignitionConfigOverride: '{{ .Spec.IgnitionConfigOverride }}'
+  nmStateConfigLabelSelector:
+    matchLabels:
+{{ if .SpecialVars.CurrentNode.HostRef }}
+      nmstate-label: "{{ .SpecialVars.CurrentNode.HostRef.Name }}"
+{{ else }}
+      nmstate-label: "{{ .SpecialVars.CurrentNode.HostName }}"
+{{ end }}
+  additionalNTPSources:
+{{ .Spec.AdditionalNTPSources | toYaml | indent 4 }}`
+
+const NodePool = `apiVersion: hypershift.openshift.io/v1beta1
+kind: NodePool
+metadata:
+  annotations:
+    siteconfig.open-cluster-management.io/sync-wave: "2"
+{{ if .SpecialVars.CurrentNode.HostRef }}
+  name: "{{ .SpecialVars.CurrentNode.HostRef.Name }}"
+  namespace: "{{ .SpecialVars.CurrentNode.HostRef.Namespace }}"
+{{ else }}
+  name: "{{ .SpecialVars.CurrentNode.HostName }}"
+  namespace: "{{ .Spec.ClusterName }}"
+{{ end }}
+{{ if .Spec.ExtraLabels.NodePool }}
+  labels:
+{{ .Spec.ExtraLabels.NodePool | toYaml | indent 4 }}
+{{ end }}
+spec:
+  clusterName: "{{ .Spec.ClusterName }}"
+  replicas: 1
```

Comment:
Can this ever be created with more than one replica to start? Should this depend on the number of nodes the user has in `ClusterInstance`?